pkg/services/services.go

```diff
@@ -218,11 +215,34 @@ func klogVerbosityArg(debug bool) string {
 }
 
 // getKubeletHostnameOverride returns the hostname override arg that should be used
-func getKubeletHostnameOverride(platformType config.PlatformType) string {
+func getKubeletHostnameOverride(platformType config.PlatformType) (string, servicescm.PowershellCmdArg) {
+	// define argument with fixed variable name
+	hostnameOverrideArg := "--hostname-override=" + hostnameOverrideVarName
 	switch platformType {
 	case config.AWSPlatformType:
-		return "--hostname-override=" + ec2HostnameVar
+		return hostnameOverrideArg, getAWSMetadataHostnamePowershellCmdArg(hostnameOverrideVarName)
+	case config.GCPPlatformType:
+		return hostnameOverrideArg, getGCPMetadataHostnamePowershellCmdArg(hostnameOverrideVarName)
 	default:
-		return ""
+		return "", servicescm.PowershellCmdArg{}
+	}
+}
+
+// getAWSMetadataHostnamePowershellCmdArg returns the PowershellCmdArg to resolve the hostname using the instance
+// metadata service in AWS. Uses AWS Tools for PowerShell (https://docs.aws.amazon.com/powershell/latest/userguide/pstools-welcome.html)
+func getAWSMetadataHostnamePowershellCmdArg(variableName string) servicescm.PowershellCmdArg {
+	return servicescm.PowershellCmdArg{
+		Name: variableName,
+		Path: "Get-EC2InstanceMetadata -Category LocalHostname",
```

Comment:
The helper should only deal with the command not the whole `PowershellCmdArg` object. 
`GetAWSHostnameCmd` is succinct and sufficiently informs us what the helper does.

Reply:
Makes sense.